app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt

```diff
@@ -45,34 +43,31 @@ class DefaultTasksRepository(
 
     override suspend fun getTasks(forceUpdate: Boolean): Result<List<Task>> {
 
-       wrapEspressoIdlingResource {
-
-           return withContext(ioDispatcher) {
-               // Respond immediately with cache if available and not dirty
-               if (!forceUpdate) {
-                   cachedTasks?.let { cachedTasks ->
-                       return@withContext Success(cachedTasks.values.sortedBy { it.id })
-                   }
-               }
+        return withContext(ioDispatcher) {
+            // Respond immediately with cache if available and not dirty
+            if (!forceUpdate) {
+                cachedTasks?.let { cachedTasks ->
+                    return@withContext Success(cachedTasks.values.sortedBy { it.id })
+                }
+            }
 
-               val newTasks = fetchTasksFromRemoteOrLocal(forceUpdate)
+            val newTasks = fetchTasksFromRemoteOrLocal(forceUpdate)
 
-               // Refresh the cache with the new tasks
-               (newTasks as? Success)?.let { refreshCache(it.data) }
+            // Refresh the cache with the new tasks
+            (newTasks as? Success)?.let { refreshCache(it.data) }
```

Comment:
I think this logic can be simpler. It's hard to follow when it's been a success and when it hasn't been. It took me a bit of time to go through the code and realise that from line 57 to 67 is the happy path (even having to cast it to Success twice) and only the last line covers the sad path. I think it'd be more understandable if in line 57 there's an if-else statement depending on the Result state.

Reply:
Comes from the master branch, so let's keep it the same for now and we can fix master later.